GiniVision/Classes/Core/Logger.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Logger.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 5/14/18.
+//
+
+import Foundation
+
+enum LogEvent: String {
+    case error = "❌"
+    case success = "✅"
+    case warning = "⚠️"
+}
+
+func Log(message: String,
+         event: LogEvent,
+         giniConfig: GiniConfiguration = .shared) {
+
+    Log(message: message, event: event.rawValue, giniConfig: giniConfig)
+}
+
+func Log(message: String,
+         event: String,
+         giniConfig: GiniConfiguration = .shared) {
+    
+    if giniConfig.debugModeOn {
+        giniConfig.logger.log(message: message, event: event)
+    }
+}
+
+@objc public protocol GiniLogger: class {
+    
+    /**
+     Logs a message
+     
+     - parameter message: Message printed out
+     - parameter event: String which represents the printed message. i.e: ⚠️
+     
+     */
+    func log(message: String, event: String)
```

Comment:
I'd avoid having partners implement a method with two parameters. If they have a custom logging solution, they probably have a method that takes one parameters. To also take an event type, they will also have to write an additional method to comply. Whereas, if it was only one, they'd likely just need to formally "say" that their logger complies to `GiniLogger `

Reply:
Then I will concat both string internally.